Modules/_datetimemodule.c

```diff
@@ -1664,6 +1664,13 @@ multiply_float_timedelta(PyObject *floatobj, PyDateTime_Delta *delta)
     ratio = _PyObject_CallMethodId(floatobj, &PyId_as_integer_ratio, NULL);
     if (ratio == NULL)
         goto error;
+    if (!PyTuple_Check(ratio)) {
```

Comment:
This is not enough. The size of the tuple should be 2.

Perhaps the code can be shared in `multiply_float_timedelta()` and `divide_timedelta_int()`.

Reply:
agh, of course.

Reply:
how can code be shared between these two functions? (I wrote a helper-function to share my patch's code between multiply_float_timedelta() and truedivide_timedelta_float().)

Reply:
The code of `multiply_float_timedelta()` and `divide_timedelta_int()` is almost the same. It is enough a one boolean parameter to distinguish multiplication from division. All code can be moved in a separate function, and `multiply_float_timedelta()` and `divide_timedelta_int()` will call it with additional argument 0 or 1.

Usually a refactoring is made only in develop version, but I think this one can be done in a bugfix change. It is enough simple and can help to fix other bugs if they will be found. What are your thoughts @abalkin?